client/my-sites/upgrades/domain-management/email-forwarding/email-forwarding-add-new.jsx

```diff
@@ -49,30 +60,40 @@ const EmailForwardingAddNew = React.createClass( {
 		event.preventDefault();
 
 		if ( this.state.formSubmitting ) {
-			return false;
+			return;
 		}
 
 		this.setState( { formSubmitting: true } );
 
-		upgradesActions.addEmailForwarding( this.props.selectedDomainName, this.state.mailbox, this.state.destination, ( error ) => {
-			this.setState( { formSubmitting: false } );
-
-			this.recordEvent( 'addNewEmailForwardClick', this.props.selectedDomainName, this.state.mailbox, this.state.destination, ! Boolean( error ) );
-
-			if ( error ) {
-				notices.error( error.message );
-			} else {
-				this.props.successNotice( this.translate( 'Yay, %(email)s has been successfully added!', { args: {
-					email: this.state.mailbox + '@' + this.props.selectedDomainName
-				} } ) );
-
-				this.clearForm();
+		this.formStateController.handleSubmit( ( hasErrors ) => {
+			if ( hasErrors ) {
+				return;
 			}
+
+			const { mailbox, destination } = formState.getAllFieldValues( this.state.fields );
+
+			upgradesActions.addEmailForwarding( this.props.selectedDomainName, mailbox, destination, ( error ) => {
+				this.recordEvent( 'addNewEmailForwardClick', this.props.selectedDomainName, mailbox, destination, ! Boolean( error ) );
+
+				if ( error ) {
+					notices.error( error.message );
+				} else {
+					this.formStateController.resetFields( this.getInitialState().fields );
+
+					notices.success(
+						this.translate( 'Yay, %(email)s has been successfully added!', { args: {
+							email: mailbox + '@' + this.props.selectedDomainName
+						} } ), {
+							duration: 5000
+						} );
+				}
+				this.setState( {formSubmitting: false, showForm: ! Boolean( error )} );
```

Comment:
Just `! error` should be enough here. Also: missing space before `formSubmitting` and at the end. Atrocious, I know - how could this even compile without those spaces?! :grin: